src/mcode-pilot/utils/filterTreatmentData.js

```diff
@@ -103,7 +82,12 @@ function generateSimilarPatientTreatments(similarPatients) {
 
     similarPatients.forEach(({ treatments }) => {
         treatments.forEach(treatment => {
-            similarPatientTreatments[treatment] = { key: treatment, name: TREATMENT_NAMES[treatment] };
+            if (typeof treatment === "object") {
+                similarPatientTreatments[treatment.code] = { key: `${treatment.code},${treatment.codeSystem}`, name: treatment.displayName, reference: treatment};
```

Comment:
shouldn't the key and the similarPatientTreatments key be the same.  There is a small chance of code being applicable across codesystems which would be an issue